### PR TITLE
Update MKL.jl to use the ILP64 interface 

### PR DESCRIPTION
**Pull request by amontoison**
It should solve the issue with MKLSparse.jl.
https://github.com/JuliaSparse/MKLSparse.jl/issues/36#issuecomment-2096148444

### CODE REVIEW

File: src/MKL.jl [hunk 1]
```diff
@@ -43,7 +43,7 @@ function lbt_forward_to_mkl()
     # MKL 2022 and onwards have 64_ for ILP64 suffixes. The LP64 interface
     # includes LP64 APIs for the non-suffixed symbols and ILP64 API for the
     # 64_ suffixed symbols. LBT4 in Julia is necessary for this to work.
-    set_interface_layer(INTERFACE_LP64)
+    set_interface_layer(BASE.USE_BLAS64 ? INTERFACE_ILP64 : INTERFACE_LP64)
```
Comment (amontoison) [inline src/MKL.jl:46]:
```suggestion
    set_interface_layer(Base.USE_BLAS64 ? INTERFACE_ILP64 : INTERFACE_LP64)
```

**Comment by amontoison** [comment]
Can we also use `INTERFACE_LP64` in `MKLSparse.jl`?

**Comment by ViralBShah** [comment]
It may be better for MKLSparse.jl to depend on MKL.jl, which can be counted upon to set the environment up correctly.

**Comment by ViralBShah** [comment]
It looks it should be safe to set the ILP64 interface. I see the symbols in the ilp64 MKL libraries.

**Review by ViralBShah — APPROVED** [review]

**Comment by amontoison** [comment]
Are you sure that symbols without suffix `_64` are still the LP64 interface?
It could break packages compiled with LBT and a LP64 BLAS.

**Comment by ViralBShah** [comment]
Darn that's a good point. They probably aren't.

**Pull request merged**